DashAI/back/routers/datasets.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+import pydantic
+from fastapi import APIRouter, File, Form, UploadFile, status
+from fastapi.encoders import jsonable_encoder
+from fastapi.exceptions import HTTPException
+from sqlalchemy import exc
+
+from DashAI.back.database import db, models
+from DashAI.back.dataloaders.classes.csv_dataloader import CSVDataLoader
+
+# from Dataloaders.classes.audioDataLoader import AudioDataLoader
+# from Dataloaders.classes.csvDataLoader import CSVDataLoader
+# from Dataloaders.classes.imageDataLoader import ImageDataLoader
+from DashAI.back.dataloaders.classes.dataloader_params import DatasetParams
+from DashAI.back.models.classes.getters import get_model_params_from_task
+from DashAI.back.routers.session_class import session_info
+from DashAI.back.tasks.task import Task
+
+router = APIRouter()
+
+
+def parse_params(params):
+    """
+    Parse JSON from string to pydantic model
+    """
+    try:
+        model = DatasetParams.parse_raw(params)
+        return model
+    except pydantic.ValidationError as e:
+        raise HTTPException(
+            detail=jsonable_encoder(e.errors()),
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+        ) from e
+
+
+@router.get("/dataset/")
+async def get_dataset():
+    """
+    Returns all the available datasets in the DB.
+    """
+    available_datasets = {}
+    try:
+        for db_dataset in db.session.query(models.Dataset):
+            available_datasets[db_dataset.id] = {
+                "dataset_name": db_dataset.name,
+                "dataset_task_name": db_dataset.task_name,
+                "dataset_path": db_dataset.path,
+            }
+    except exc.SQLAlchemyError:
+        return {"message": "Couldn't connect with DB."}
+    return available_datasets
+
+
+@router.post("/dataset/")
+async def upload_dataset(
+    params: str = Form(), url: str = Form(None), file: UploadFile = File(None)
```

Comment:
What syntax are you thinking about? The next line calls a pydantic model who raises an exception if the form validation fails
```
params = parse_params(params)
```

Reply:
Try params: DatasetParams

Reply:
The problem is that cannot upload the pydantic model with a file in the same endpoint, for that is necessary this 'trick'. Other solution is divide this endpoint in two, one for upload the parameters directly with the pydantic model and other for upload the file and process the dataset.

Reply:
Ok, I didn't know, thanks!

Reply:
> Other solution is divide this endpoint in two

I don't like this approach because it brings more problems than solutions, e.g: if one endpoint fails we need to communicate that the other, etc